tom/tom.py

```diff
@@ -90,40 +82,44 @@ def privmsg(self, user, channel, message):
+            self.msg(self.channel, "ADTRAN, Inc. is a provider of telecommunications networking equipment and "
+                                   "internetworking products. Its headquarters are in Huntsville, Alabama.")
+        elif search(r"(\bidea\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "That's a great idea!!")
-        elif search(r'(^|\s)+hackathon *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\bhackathon\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "Woooo! Hackathon!!!!")
-        elif search(r'(^|\s)+adtran *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\badtran\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "Goooooo ADTRAN!") 
-        elif search(r'(^|\s)+tom *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\btom\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "It's me! Tom!!! On the eighth floor!!!!") 
-        elif search(r'(^|\s)+TA5K *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\bta5k\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "You should go with the flamethrower attachment") 
-        elif search(r'(^|\s)+TA5Ks *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\bta5ks\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "our pride and joy the TA5K")
-        elif search(r'(^|\s)+r-olt *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\b(rolt|r-olt)\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "Tomothy Remote-Optical-Line-Terminal Stanton is the name, don't wear it out ;)")
-        elif search(r'(^|\s)+dynetics *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\bdynetics\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "no cursing in adtran go wash your mouth")
-        elif search(r'(^|\s)+work *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\bwork\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "werk werk werk werk")
-        elif search(r'(^|\s)+Calix *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\bcalix\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "no sir")
-        elif search(r'(^|\s)+working *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\bworking\b)", message, IGNORECASE):
             self.currentTime = timeRightNow
             self.msg(self.channel, "keep working hard!!!!")
-        elif search(r'(^|\s)+song *(\s|$)+', message, IGNORECASE):
+        elif search(r"(\bfight\ssong\b)", message, IGNORECASE):
```

Comment:
very nice regex. had no clue.

Reply:
Thank Jimmy; I looked at memebot since it has to parse all sorts of conditions.